app/src/main/java/com/android/shelfLife/model/invitations/InvitationViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package com.android.shelfLife.model.invitations
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import com.android.shelfLife.model.household.HouseHold
+import com.google.firebase.auth.FirebaseAuth
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class InvitationViewModel(private val invitationRepository: InvitationRepository) : ViewModel() {
+
+  private val _invitations = MutableStateFlow<List<Invitation>>(emptyList())
+  val invitations: StateFlow<List<Invitation>> = _invitations.asStateFlow()
+
+  init {
+    FirebaseAuth.getInstance().addAuthStateListener { firebaseAuth ->
+      if (firebaseAuth.currentUser != null) {
+        startListeningForInvitations()
```

Comment:
Previously you only listened when the app was opened. Now with this it will constantly listen.